ui/components/app/edit-gas-fee-popover/network-status/network-status.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import { COLORS } from '../../../../helpers/constants/design-system';
+import { useGasFeeContext } from '../../../../contexts/gasFee';
+import I18nValue from '../../../ui/i18n-value';
+import Typography from '../../../ui/typography/typography';
+
+import StatusSlider from './status-slider';
+
+const NetworkStatus = () => {
+  const { gasFeeEstimates } = useGasFeeContext();
+
+  return (
+    <div className="network-status">
+      <Typography
+        color={COLORS.UI4}
+        fontSize="10px"
+        fontWeight="bold"
+        margin={[3, 0]}
+        variant="h6"
+      >
+        <I18nValue messageKey="networkStatus" />
+      </Typography>
+      <div className="network-status__info">
+        <div className="network-status__info-field">
+          <span className="network-status__info-field-data">
+            {gasFeeEstimates?.estimatedBaseFee} GWEI
```

Comment:
Should we truncate this number? Perhaps to 2 or 3 decimal places?

Also, if `gasFeeEstimates` is null then this will just appear as "GWEI". Should we even render this component unless we are guaranteed to have `gasFeeEstimates`? Or should we render some kind of loader that will appear until we have `gasFeeEstimates`?

Reply:
I added condition to show edit-gas-fee modal only if estimates are present. 

edit-gas-fee modal does show `LoadingHeartBeat` component while gas estimates are loading.

Reply:
Okay. So if the modal only appears if `gasFeeEstimates` then do you need the null check here? I get a little sensitive about null checks because it means that we don't trust the upstream code, whatever that is. From a type perspective, it seems that either `gasFeeEstimates` is null or that it's something. But correct me if I'm wrong.

Reply:
LoadingHeartBeat does not hide edit gas modal content but shows a loading effect. Thus check for null is needed to avoid exception being thrown.

Reply:
Hmm ok. Just feels like we could do this better. I don't have enough familiarity to know what that is yet though. More food for thought I guess.